app/models/game.rb

```diff
@@ -38,8 +37,55 @@ def populate_game!
     end
   end
 
+  def in_check?
+    kings = self.pieces.find_by(type: 'King')
+    
+    kings.each.do |king|
+      if king.color == "black"
+        black_in_check?
+      else 
+        white_in_check?
+      end 
+    end 
+  end 
+
+  def black_in_check?
+    black_king = self.pieces.find_by(type: 'King', color: 'black')
+    position_x = black_king.position_x
+    position_y = black_king.position_y
+
+    self.pieces.each do |piece|
+      if piece.valid_move?(position_x, position_y) && piece.color == 'white'
```

Comment:
instead if/else you could just say:
`return true if piece.valid_move?(position_x, position_y) && piece.color == 'white'`

Reply:
we could also have sth like `white_pieces = ...` and iterate over them, saving the `&& piece.color`